intake/catalog/local.py

```diff
@@ -623,14 +631,16 @@ def _load(self):
         if isinstance(self.path, (list, tuple)):
             files = sum([open_files(p, mode='rb', **options)
                          for p in self.path], [])
-            self.name = "%i files" % len(files)
+            self.name = self.name or "%i files" % len(files)
+            self.description = self.description or f'Catalog generated from {len(files)} files'
             self.path = [make_path_posix(p) for p in self.path]
         else:
             if isinstance(self.path, str) and '*' not in self.path:
                 self.path = self.path + '/*'
             files = open_files(self.path, mode='rb', **options)
             self.path = make_path_posix(self.path)
-            self.name = self.path
+            self.name = self.name or self.path
+            self.description = self.description or f'Catalog generated from all files found in {self.path}'
```

Comment:
*here* perhaps we can just have `self.path` and be done. In the repr, you get the word catalog anyway.

Reply:
Actually, changed my mind - forget it.